### PR TITLE
Aggregations now do not draw if no documents are present

### PR DESCRIPTION
**Pull request by corrigancd**
Fix for https://github.com/sirensolutions/kibi-internal/issues/9548

### CODE REVIEW

File: public/POIs.js [hunk 1]
```diff
@@ -4,13 +4,16 @@ import { markerIcon } from 'plugins/enhanced_tilemap/vislib/markerIcon';
 import { toLatLng } from 'plugins/enhanced_tilemap/vislib/geo_point';
 import { SearchSourceProvider } from 'ui/courier/data_source/search_source';
 import { FilterBarQueryFilterProvider } from 'ui/filter_bar/query_filter';
+import { GeoFilterFactory } from 'plugins/enhanced_tilemap/vislib/geoFilter';
 import utils from 'plugins/enhanced_tilemap/utils';
+import { PassThrough } from 'stream';
```
Comment (szydan) [inline public/POIs.js]: not used remove

File: public/POIs.js [hunk 2]
```diff
@@ -40,26 +43,56 @@ define(function (require) {
...
         }
         searchSource.size(this.limit);
         searchSource.source({
           includes: _.compact(_.flatten([this.geoField, this.popupFields])),
           excludes: []
         });
+
+        const tooManyDocsInfo = [
+          '<i class="fa fa-exclamation-triangle text-color-warning doc-viewer-underscore"></i>',
+          '<b><p class="text-color-warning">Caution: Undisplayed POIs present on map canvas,<br>' +
```
Comment (szydan) [inline public/POIs.js]: for multiline strings you can use the javascriot template strings  (backtics)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: public/visController.js [hunk 3]
```diff
@@ -101,12 +101,26 @@ define(function (require) {
...
-        size: _.get(layerParams, 'markerSize', 'm')
+        size: _.get(layerParams, 'markerSize', 'm'),
+        mapExtentFilter: {
+          geo_bounding_box: getGeoBoundingBox(),
+          geoField: getGeoField()
+        }
       };
+
+      //Element rendered in Leaflet Library
+      options.$legend = $element.find('a.leaflet-control-layers-toggle');
```
Comment (szydan) [inline public/visController.js]: I think there can not be more than one legend so maybe get it here like 
```
$legend = $element.find('a.leaflet-control-layers-toggle').get(0);
if ($legend ) {
options.$legend = $leggend
```
the you will be able to drop the [0] from other places

File: public/POIs.js [hunk 4]
```diff
@@ -40,26 +43,56 @@ define(function (require) {
...
-        .then(searchResp => {
-          callback(self._createLayer(searchResp.hits.hits, geoType, options));
-        });
+          .then(searchResp => {
+
+            //Too many documents warning for each specific layer
+            options.$legend[0].tooManyDocsInfo = '';
+
+            if (searchResp.hits.total > this.limit) {
+              options.$legend[0].tooManyDocsInfo = tooManyDocsInfo;
```
Comment (szydan) [inline public/POIs.js]: Why do we need to set "tooManyDocsInfo" on the legend element?
Could you explain it, please

Comment (corrigancd) [reply public/POIs.js]: For consistency I guess and also to ensure that what I was doing wasn't affecting anything else. I've removed the $legend element and it seems to have the same functionality as with it in place.

File: public/visController.js (hunk 3 above)
Comment (corrigancd) [reply public/visController.js]: nice, changed.

File: public/POIs.js (hunk 4 above)
Comment (corrigancd) [reply public/POIs.js]: Ok, closer look at this. The warning triangle appears next to the layer without this. But the $legend element is the link to the "a.leaflet-control-layers-toggle" element in Leaflet library. The warning icon won't appear leke below if $legend reference is removed. 

image

Comment (corrigancd) [reply public/POIs.js]: 1 below is the $legend element when there are less POIs present than the limit
2 below is the $legend element when there are more POIs than the limit. 

$legend is a link to the html that is used within the leaflet library to make the warning icon appear 

image

**Pull request merged**